implementations/python/mzlib/backends/msp.py

```diff
@@ -155,10 +163,28 @@ def _get_peptide_sequence_for_analyte(self, spectrum, analyte_reference=None):
         analyte = spectrum.analytes.get(analyte_reference)
         if analyte is None:
             return None
-        return analyte.get_attribute('MS:1000888|unmodified peptide sequence')
+        return analyte.get_attribute('MS:1000888|stripped peptide sequence')
 
 
 parse_annotation = MSPAnnotationStringParser(annotation_pattern)
+modification_list_parser = re.compile(r"(\d+),([ARNDCEQGHKMFPSTWYVIL]),([A-Za-z0-9]+)")
+
+
+def parse_modification_notation(text: str) -> List[Tuple[int, str, str]]:
+    if not isinstance(text, str):
+        return []
+    i = 0
+    n = len(text)
+
+    mods = []
+    while text[i].isdigit() and i < n:
+        i += 1
+
+    for position, residue, mod in modification_list_parser.findall(text):
+        position = int(position)
+        modification_name = modification_name_map[mod]
```

Comment:
This map now only contains carbamidomethyl. How about `modification_name_map.get(mod, mod)` to use the mod value as fallback when no mapping is available?

Reply:
A good idea. I had left it to break noisily when it encountered a new modification, but I rewrote this block to more explicitly talk about finding a modification it doesn't know yet.